src/Storage/Device/S3.php

```diff
@@ -147,14 +147,18 @@ public function getRoot(): string
      *
      * @return string
      */
-    public function getPath(string $filename): string
+    public function getPath(string $filename, $prefix = null): string
```

Comment:
The docblock needs to be updated for this addition.

Additionally can we add the datatype to the argument?